api/receiver/appengine.go

```diff
@@ -57,12 +58,21 @@ func NewAppEngineAPI(ctx context.Context) AppEngineAPI {
 	return &appEngineAPIImpl{
 		AppEngineAPIImpl: shared.NewAppEngineAPI(ctx),
 		queue:            ResultsQueue,
+		store:            shared.NewAppEngineDatastore(ctx, false),
 	}
 }
 
 func (a *appEngineAPIImpl) addTestRun(testRun *shared.TestRun) (*DatastoreKey, error) {
-	key := datastore.NewIncompleteKey(a.Context(), "TestRun", nil)
-	key, err := datastore.Put(a.Context(), key, testRun)
+	ctx := a.Context()
+	var key shared.Key
+	var err error
+	store := shared.NewAppEngineDatastore(ctx, false)
```

Comment:
Use `a.store` instead of creating another one.

Reply:
Done.